### PR TITLE
feat: define protobuf interface to communicate with ota-client

### PR DESCRIPTION
**Pull request by airkei**
### Why
https://tier4.atlassian.net/browse/RT4-13461
To retrieve the metrics, we plan to refactor the interface between ota-client and otaclient-iot-logging-server from REST to gRPC based on [this design](https://tier4.atlassian.net/wiki/spaces/OTA/pages/3481894920/OTA+Metrics).

### What
This PR just defines the protobuf interface to communicate.
Core and test part will be implemented in the later PR.

### Tests
Verified that `hatch build -t wheel` creates the expected `.whl`.

### CODE REVIEW

File: proto/otaclient_iot_logging_server_v1.proto [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+   */
+  rpc PutLog(PutLogRequest) returns (PutLogResponse) {}
+}
+
+enum LogType {
+  LOG = 0;
+  METRICS = 1;
+}
+enum LogLevel {
+  TRACE = 0;
```
Comment (Bodong-Yang) [inline proto/otaclient_iot_logging_server_v1.proto:29]: Protobuf by default will use 0 value for unset field, I think we should reserve 0 value to `UNSPECIFIC` here.

File: proto/otaclient_iot_logging_server_v1.proto [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  DEBUG = 1;
+  INFO = 2;
+  WARN = 3;
+  ERROR = 4;
+  FATAL = 5;
+}
+enum ErrorCode {
+  OK = 0;
+  UNKNOWN = 1;
```
Comment (Bodong-Yang) [inline proto/otaclient_iot_logging_server_v1.proto:38]: What is the meaning of `UNKNOWN` here 🤔. I think `FAILED` might be more proper here.

**Comment by Bodong-Yang** [comment]
I left some comments related to proto definition, basically I think there is no problem.

File: proto/otaclient_iot_logging_server_v1.proto (hunk 1 above)
Comment (airkei) [reply proto/otaclient_iot_logging_server_v1.proto]: Thank you, the original idea was to fallback to `TRACE` if the level is not set. Added `UNSPECIFIC` as 0.

**Comment by Bodong-Yang** [comment]
LGTM!

File: proto/otaclient_iot_logging_server_v1.proto (hunk 2 above)
Comment (airkei) [reply proto/otaclient_iot_logging_server_v1.proto]: Good point. I plan to increase the error code as necessary. But `UNKNOWN` is very ambiguous word. Not sure if it was OK, FAILED. Fixed it.

**Review by Bodong-Yang — APPROVED** [review]

**Pull request merged**